### ISSUE TITLE
Allow chat history (selection) to be stored with a file/folder as comments

### ISSUE THREAD

**Issue by putt1ck** [nextcloud/spreed#232]
As per comments on https://github.com/nextcloud/spreed/issues/35:
https://github.com/nextcloud/spreed/issues/35#issuecomment-274319165
&
https://github.com/nextcloud/spreed/issues/35#issuecomment-274721974

It would be useful to have the ability to export/copy a text chat history or part thereof and have it associated with a file or folder that is created as a result of the discussion, possibly as comments.

**Comment by nickvergessen** [comment]
Should be moved to the chat app: https://github.com/jsxc/jsxc.nextcloud

**Comment by bes1002t** [comment]
But imho this should be also encrypted on the file system.

**Comment by shankari** [comment]
It would be great if there would be a way to automate this via a script so that we can include standard text logs in our nightly backups. Yes, we backup the database too, but databases are a lot less portable and prone to corruption

**Comment by notklaatu** [comment]
For others looking for this functionality, you can get your logs this way:

    $ mysql -u $DB_USER -p -h $HOST -D $DB -e 'SELECT * FROM oc_comments;' > ~/chat_`date +%d.%Y`.log

**Comment by shankari** [comment]
@notklaatu this is will have duplicate entries in consecutive logs, no?

So if I have chat that extended from 1st Jan to 7th Jan
- `chat_2019.01` will have the messages from 1st Jan
- `chat02;.2019.02` will have messages from both 1st and 2nd Jan
- `chat03.2019` will have messages for 1st, 2nd and third


That seems unintuitive.

**Comment by nickvergessen** [comment]
The command from @notklaatu also adds all chats into one file.

**Comment by wiswedel** [comment]
> The command from @notklaatu also adds all chats into one file.

And keep in mind: It only gets chats from rooms that still exist. Deleting a room also deletes all of its chat history from the database.

**Comment by eekhof** [comment]
it would be really nice to have this, if I can help I wil

**Comment by j-j-v** [comment]
agreed - can model it on IRC logs (timestamp/user/message/(...) per line, daily file with day's date for a  filename in per conversation folder - at a sketch) & make available to users via a link & ideally a built-in search, maybe with "from:username" & such niceties.

Thanks!

**Comment by wbob** [comment]
I needed a minimal history export on a whim and came up with:
```
select r.name,c.creation_timestamp,concat(c.actor_id, ': ',c.message)
  from oc_comments c inner join oc_talk_rooms r on c.object_id = r.id
 where c.object_type = 'chat'
 order by r.id,c.creation_timestamp desc;
```
JSON (mysql/mariadb)
```
select json_pretty(json_merge(
  json_object('room',r.name),
  json_object('timestamp',c.creation_timestamp),
  json_object('name',c.actor_id),
  json_object('message',c.message)
)) from oc_comments c inner join oc_talk_rooms r on c.object_id = r.id
where c.object_type = 'chat'
order by r.id,c.creation_timestamp desc;
```
echo pipe this to `mysql -BN` to get the extract

**Comment by Gekko23** [comment]
So, EIGHT YEARS after the initial request we're still echo piping to get an essential feature, cause noone at Nextcloud thinks exporting talk history is a thing? Even within the "user migration" addon this is still not solved.
